distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc

```diff
@@ -0,0 +1,378 @@
+
+=== Developing Catalog Plugins
+
+Plugins extend the functionality of the Catalog Framework by performing actions at specified times during a transaction. 
+
+By implementing a plugin interface, actions can be performed at the desired time. 
+
+The following types of plugins can be created:
+
+.Plugin Interfaces
+[cols="1,2m,1,1", options="header"]
+|===
+
+|Plugin Type
+|Plugin Interface
+|Description
+|Example
+
+|Pre-Ingest
+|${ddf-branding}.catalog.plugin.PreIngestPlugin
+|Runs before the Create/Update/Delete method is sent to the CatalogProvider
+|Metadata validation services
+
+|Post-Ingest
+|${ddf-branding}.catalog.plugin.PostIngestPlugin
+|Runs after the Create/Update/Delete method is sent to the CatalogProvider
+|EventProcessor for processing and publishing event notifications to subscribers
+
+|Pre-Query
+|${ddf-branding}.catalog.plugin.PreQueryPlugin
+|Runs prior to the Query/Read method being sent to the Source
+|An example is not included with ${branding}
+
+|Post-Query
+|${ddf-branding}.catalog.plugin.PostQueryPlugin
+|Runs after results have been retrieved from the query but before they are posted to the Endpoint
+|An example is not included with ${branding}
+
+|Pre-Subscription
+|${ddf-branding}.catalog.plugin.PreSubscription
+|Runs prior to a Subscription being created or updated
+|Modify a query prior to creating a subscription
+
+|Pre-Delivery
+|${ddf-branding}.catalog.plugin.PreDeliveryPlugin
+|Runs prior to the delivery of a Metacard when an event is posted
+|Inspect a metacard prior to delivering it to the Event Consumer
+
+|Pre-Resource
+|${ddf-branding}.catalog.plugin.PreResource
+|Runs prior to a Resource being retrieved
+|An example is not included with ${branding}
+
+|Post-Resource
+|${ddf-branding}.catalog.plugin.PostResource
+|Runs after a Resource is retrieved, but before it is sent to the Endpoint
+|Verification of a resource prior to returning to a client
+
+|Policy
+|${ddf-branding}.catalog.plugin.PolicyPlugin
+|Runs prior to all other catalog plugins to esablish the policy for requests/responses
+|An example is MetacardValidityFilterPlugin
+
+|Access
+|${ddf-branding}.catalog.plugin.AccessPlugin
+|Runs directly after the PolicyPlugin
+|An examples are the FilterPlugin and OperationPlugin
+
+|===
+
+==== Implement Plugins
+
+The procedure for implementing any of the plugins follows a similar format:
```

Comment:
Might want to add a note about performance and threading concerns. One bad plugin can tank the system.